src/RenderView.tsx

```diff
@@ -141,10 +182,20 @@ export default function RenderView(props: Props) {
 				cameraControls.moveForward(-moveMeters * scaleFactor);
 			else
 				cameraControls.moveForward(moveMeters * scaleFactor);
+			setCamPosX(camera.position.x);
+			setCamPosY(camera.position.y);
+			setCamPosZ(camera.position.z);
 		}
 		/* Look controls */
 		const mouseDownHandler = (e: MouseEvent) => {if (e.button===1) cameraControls.lock();}; //capture mouse movement (to rotate view) while holding middle click
-		const mouseUpHandler = (e: MouseEvent) => {if (e.button===1) cameraControls.unlock();}; //release mouse after middle click is over
+		const mouseUpHandler = (e: MouseEvent) => { //release mouse after middle click is over
+			if (e.button===1) {
+				cameraControls.unlock();
+				setCamRotX(camera.rotation.x);
```

Comment:
moving this only on unlock feels janky, we could probably capture mousemove too?

Reply:
I wasn't quite sure at the time how I could get it to follow the mouse, but I figured it out. It should be good now.